EIPS/eip-5058.md

```diff
@@ -0,0 +1,185 @@
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Lockable ERC-721 **MUST** implement the `IERC721Lockable` interfaces:
+```solidity
+// SPDX-License-Identifier: MIT
+// Creator: tyler@radiocaca.com
+
+pragma solidity ^0.8.8;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+/**
+ * @dev ERC-721 Non-Fungible Token Standard, optional lockable extension
+ * ERC721 Token that can be locked for a certain period and cannot be transferred.
+ * This is designed for a non-escrow staking contract that comes later to lock a user's NFT
+ * while still letting them keep it in their wallet.
+ * This extension can ensure the security of user tokens during the staking period.
+ * If the nft lending protocol is compatible with this extension, the trouble caused by the NFT
+ * airdrop can be avoided, because the airdrop is still in the user's wallet
+ */
+interface IERC721Lockable is IERC165 {
+    /**
+     * @dev Emitted when `tokenId` token is locked from `from`.
+     */
+    event Locked(address indexed operator, address indexed from, uint256 indexed tokenId, uint256 expired);
+
+    /**
+     * @dev Emitted when `tokenId` token is unlocked from `from`.
+     */
+    event Unlocked(address indexed operator, address indexed from, uint256 indexed tokenId);
+
+    /**
+     * @dev Emitted when `owner` enables `approved` to lock the `tokenId` token.
+     */
+    event LockApproval(address indexed owner, address indexed approved, uint256 indexed tokenId);
+
+    /**
+     * @dev Emitted when `owner` enables or disables (`approved`) `operator` to lock all of its tokens.
+     */
+    event LockApprovalForAll(address indexed owner, address indexed operator, bool approved);
+
+    /**
+     * @dev Returns the current locker of the `tokenId` token.
+     *
+     * Requirements:
+     *
+     * - `tokenId` must exist.
+     */
+    function lockerOf(uint256 tokenId) external view returns (address locker);
+
+    /**
+     * @dev Lock `tokenId` token until `expired` from `from`.
+     *
+     * Requirements:
+     *
+     * - `from` cannot be the zero address.
+     * - `tokenId` token must be owned by `from`.
+     * - `expired` must be greater than block.timestamp
+     * - If the caller is not `from`, it must be approved to lock this token
+     * by either {lockApprove} or {setLockApprovalForAll}.
+     *
+     * Emits a {Locked} event.
+     */
+    function lockFrom(
+        address from,
+        uint256 tokenId,
+        uint256 expired
+    ) external;
+
+    /**
+     * @dev Unlock `tokenId` token until `expired` from `from`.
```

Comment:
Why unlock it _until_ the expiration?

I'd say that it should probably remain unlocked after the lock has expired.